### PR TITLE
Add Performer coversheet generation feature

### LINKED ISSUE #557: Performer coversheet generation
Add ability for administrators to generate a coversheet to submit with invoices for performers.

### PR DESCRIPTION
**Pull request by madison-boman**
Helps user generate an excel sheet for a mail merge on all performers.
Fix #557 Performer coversheet generation

### CODE REVIEW

File: src/GRA.Controllers/GRA.Controllers.csproj [hunk 1]
```diff
@@ -38,6 +38,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="AntiXSS" Version="4.3.0" />
```
Comment (iafb) [inline src/GRA.Controllers/GRA.Controllers.csproj]: I think this is unused and can be removed?

File: src/GRA.Controllers/MissionControl/PerformerManagementController.cs [hunk 2]
```diff
@@ -2,18 +2,25 @@
...
 using GRA.Domain.Model.Filters;
 using GRA.Domain.Service;
 using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.Rendering;
 using Microsoft.Extensions.Logging;
 using Newtonsoft.Json;
+using DocumentFormat.OpenXml;
+using DocumentFormat.OpenXml.Packaging;
```
Comment (iafb) [inline src/GRA.Controllers/MissionControl/PerformerManagementController.cs]: Remove and sort usings (ctrl+r, ctrl+g) should be run to remove the duplicate

File: src/GRA.Controllers/MissionControl/PerformerManagementController.cs [hunk 3]
```diff
@@ -27,24 +34,47 @@ public class PerformerManagementController : Base.MCController
...
+            "Performer Billing"
+        };
+
         private static readonly DateTime DefaultPerformerScheduleStartTime = DateTime.Parse("8:00 AM");
         private static readonly DateTime DefaultPerformerScheduleEndTime = DateTime.Parse("8:00 PM");
 
         private readonly ILogger<PerformerManagementController> _logger;
         private readonly PerformerSchedulingService _performerSchedulingService;
         private readonly SiteService _siteService;
+        private readonly Domain.Repository.IBranchRepository _branchRepository;
```
Comment (iafb) [inline src/GRA.Controllers/MissionControl/PerformerManagementController.cs]: Repositories shouldn't be referenced in the controller, SiteService should have the needed branch calls

File: src/GRA.Controllers/MissionControl/PerformerManagementController.cs [hunk 4]
```diff
@@ -27,24 +34,47 @@ public class PerformerManagementController : Base.MCController
...
+        private readonly IHostingEnvironment _hostingEnvironment;
```
Comment (iafb) [inline src/GRA.Controllers/MissionControl/PerformerManagementController.cs]: _hostingEnvironment is unused and can be removed

File: src/GRA.Data/Model/PsSettings.cs [hunk 5]
```diff
@@ -11,6 +11,15 @@ public class PsSettings : Abstract.BaseDbEntity
...
 
+        [MaxLength(255)]
+        public string StaffContact { get; set; }
+
+        [MaxLength(255)]
+        public string FundingSource { get; set; }
+
+        [MaxLength(255)]
+        public string LibraryBranch { get; set; }
+
```
Comment (iafb) [inline src/GRA.Data/Model/PsSettings.cs]: With the change to use mail merge these properties can be removed

File: src/GRA.Data/Repository/PsBranchSelectionRepository.cs [hunk 6]
```diff
@@ -52,6 +52,14 @@ public async Task<int> GetCountByPerformerIdAsync(int performerId)
...
                 .CountAsync();
         }
+        public async Task<ICollection<PsBranchSelection>> GetByPerformerIdAsync(int performerId)
+        {
+            return await DbSet
+                .AsNoTracking()
+                .Where(_ => _.ProgramId.HasValue && _.Program.PerformerId == performerId)
+                .ProjectTo<PsBranchSelection>(_mapper.ConfigurationProvider)
+                .ToListAsync();
+        }
```
Comment (iafb) [inline src/GRA.Data/Repository/PsBranchSelectionRepository.cs]: The existing GetByPerformerIdAsync method should be able to be called without needing an overload

File: src/GRA.Controllers/MissionControl/PerformerManagementController.cs [hunk 7]
```diff
@@ -263,7 +301,8 @@ public async Task<IActionResult> Performer(int id)
                 ShowAlertDanger("Unable to view performer: ", gex);
                 return RedirectToAction(nameof(Performers));
             }
-
+            performer.SelectionsCount = await _performerSchedulingService
+                        .GetPerformerSelectionCountAsync(performer.Id);
```
Comment (iafb) [inline src/GRA.Controllers/MissionControl/PerformerManagementController.cs]: This value isn't used in the performer view and doesn't need to be assigned

File: src/GRA.Domain.Service/PerformerSchedulingService.cs [hunk 8]
```diff
@@ -567,6 +591,12 @@ public async Task<int> GetPerformerSelectionCountAsync(int performerId)
             return await _psBranchSelectionRepository.GetCountByPerformerIdAsync(performerId);
         }
 
+        public async Task<ICollection<PsBranchSelection>> GetPerformerSelectionAsync(int performerId)
+        {
+            VerifyManagementPermission();
+            return await _psBranchSelectionRepository.GetByPerformerIdAsync(performerId);
+        }
+
```
Comment (iafb) [inline src/GRA.Domain.Service/PerformerSchedulingService.cs]: GetPerformerBranchSelectionsAsync performs the same function and can be used in place of this

**Comment by k7hpn** [comment]
This was fixed in #1069.

**Pull request closed without merging**